src/NewsService.Validation/Channel/CreateChannelRequestValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.Kernel.Validators.Interfaces;
+using LT.DigitalOffice.NewsService.Data.Interfaces;
+using LT.DigitalOffice.NewsService.Models.Dto.Requests.Channel;
+using LT.DigitalOffice.NewsService.Validation.Channel.Interfaces;
+
+namespace LT.DigitalOffice.NewsService.Validation.Channel
+{
+  public class CreateChannelRequestValidator : AbstractValidator<CreateChannelRequest>, ICreateChannelRequestValidator
+  {
+    private readonly IChannelRepository _repository;
+
+    public CreateChannelRequestValidator(
+      IImageContentValidator imageContentValidator,
+      IImageExtensionValidator imageExtensionValidator,
+      IChannelRepository repository)
+    {
+      _repository = repository;
+
+      RuleFor(c => c.Name)
+        .NotEmpty().WithMessage("Name must not be empty.")
+        .MustAsync(async (request, _) => !await _repository.IsNameExistAsync(request))
```

Comment:
!await repository.IsNameExistAsync(request)

Reply:
It was my misteke to create methods IsNameExistAsync, rename to DoesNameExistAsync?